src/Turing.jl

```diff
@@ -135,21 +135,13 @@ export  @model,                 # modelling
         optim_function,
         optim_problem
 
+if !isdefined(Base, :get_extension)
+    using Requires
+end
+
 function __init__()
     @static if !isdefined(Base, :get_extension)
```

Comment:
Can you re-try to move the `@static` check outside of `__init__()`? Maybe it works now since we do not use the `@init` macro anymore.

Reply:
I am afraid it still results in compilation errors see below:

ERROR: LoadError: ArgumentError: invalid @static macro
```
Stacktrace:
 [1] var"@static"(__source__::LineNumberNode, __module__::Module, ex::Any)
   @ Base ./osutils.jl:37
 [2] include
   @ ./Base.jl:457 [inlined]
 [3] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt128}}, source::Nothing)
   @ Base ./loading.jl:2010
 [4] top-level scope
   @ stdin:2
in expression starting at /home/jaimerz/Cambridge/Turing.jl/src/Turing.jl:142
in expression starting at /home/jaimerz/Cambridge/Turing.jl/src/Turing.jl:1
in expression starting at stdin:2
```

Reply:
What exactly did you try?

Reply:
```
@static  function __init__()
    if !isdefined(Base, :get_extension)
        @require Optim="429524aa-4258-5aef-a3af-852621145aeb" include("../ext/TuringOptimExt.jl")
        @require DynamicHMC="bbc10e6e-7c05-544b-b16e-64fede858acb" include("../ext/TuringDynamicHMCExt.jl")
  end
end
```

Reply:
It should be
```julia
@static if !isdefined(Base, :get_extension)
    function __init__()
        @require Optim="429524aa-4258-5aef-a3af-852621145aeb" include("../ext/TuringOptimExt.jl")
        @require DynamicHMC="bbc10e6e-7c05-544b-b16e-64fede858acb" include("../ext/TuringDynamicHMCExt.jl")
  end
end
```
Can you try that?